### PR TITLE
Improve Unicode Handling in JSON Tooling

### PR DESCRIPTION
**Pull request by UmbralReaper**
#### Summary
None

#### Purpose of change
When working on tooling for mods it is not uncommon to discover that something doesn't work due to a problem in utf-8 handling. 

#### Describe the solution
https://github.com/CleverRaven/Cataclysm-DDA/pull/49771/commits/0316316467b9d522c7bfe543e494d0b18dabe358 - Explicitly declare encoding as utf-8 whenever a file is opened 
https://github.com/CleverRaven/Cataclysm-DDA/pull/49771/commits/cb8241f9afd24b6cef75cead9b381d37dae2e3ce - Set ensure_ascii=False wherever json.dump is called

#### Describe alternatives you've considered
Not doing this.

#### Testing
This shouldn't cause any errors.

### CODE REVIEW

File: tools/json_tools/cddatags.py [hunk 1]
```diff
@@ -72,7 +72,7 @@ def main(args):
     json_tags_lines = [make_tags_line(*d) for d in definitions]
     existing_tags_lines = []
     try:
-        with open(TAGS_FILE, 'rb') as tags_file:
+        with open(TAGS_FILE, 'rb', encoding="utf-8") as tags_file:
```
Comment (jbytheway) [inline tools/json_tools/cddatags.py:75]: FYI, you accidentally broke this because you shouldn't add an encoding option when opening in binary mode.  I've PRed a fix as part of some unrelated changes in #49980 where I noticed it.

Comment (UmbralReaper) [reply tools/json_tools/cddatags.py:75]: Oops, I don't know how I missed that 😅. Thanks for the fix!
